src/crypto/byron/ledgerUtils.js

```diff
@@ -0,0 +1,414 @@
+// @flow
+
+import AppAda, {ErrorCodes} from '@cardano-foundation/ledgerjs-hw-app-cardano'
+import {BigNumber} from 'bignumber.js'
+import TransportBLE from '@ledgerhq/react-native-hw-transport-ble'
+import {TransportStatusError} from '@ledgerhq/hw-transport'
+import {BleError} from 'react-native-ble-plx'
+import ExtendableError from 'es6-error'
+
+import {Logger} from '../../utils/logging'
+import {CONFIG} from '../../config'
+import {
+  generateFakeWallet,
+  signTransaction,
+  encodeTxAsRust,
+  decodeRustTx,
+} from './util'
+import {Bip32PublicKey} from 'react-native-chain-libs'
+
+import type {
+  Addressing,
+  PreparedTransactionData,
+  TransactionInput,
+  TransactionOutput,
+  TxBodiesResponse,
+  V1SignedTx,
+} from '../../types/HistoryTransaction'
+import type {
+  GetVersionResponse,
+  GetExtendedPublicKeyResponse,
+  InputTypeUTxO,
+  OutputTypeAddress,
+  OutputTypeChange,
+  SignTransactionResponse,
+  Witness,
+} from '@cardano-foundation/ledgerjs-hw-app-cardano'
+import type {TxWitness} from './util'
+import type BluetoothTransport from '@ledgerhq/react-native-hw-transport-ble'
+
+//
+// ============== Errors ==================
+//
+
+export class BluetoothDisabledError extends ExtendableError {
+  constructor() {
+    super('BluetoothDisabledError')
+  }
+}
+
+export class GeneralConnectionError extends ExtendableError {
+  constructor() {
+    super('GeneralConnectionError')
+  }
+}
+
+export class LedgerUserError extends ExtendableError {
+  constructor() {
+    super('LedgerUserError')
+  }
+}
+
+const _isConnectionError = (e: Error | TransportStatusError): boolean => {
+  if (
+    e instanceof BleError ||
+    e.message.includes('was disconnected') ||
+    e.message.includes('DisconnectedDevice') ||
+    e.name.includes('DisconnectedDevice') ||
+    e.message.includes('not found')
+  ) {
+    return true
+  }
+  return false
+}
+
+// note: e.statusCode === ErrorCodes.ERR_CLA_NOT_SUPPORTED probably means
```

Comment:
> probably

Have we confirmed this, or are we leaving it as "probably" for now?

Reply:
Yes, "probably" is not precisely what I meant. Theoretically that error rises when the wrong app is opened in ledger, but I've observed it a few times when no app is opened at all and a tx is requested. So I said "probably" because the error is "probably" due to the user not having opened the ADA app instead of having opened the wrong app.